ports/stm32/machine_i2s.c

```diff
@@ -730,7 +746,12 @@ STATIC void machine_i2s_init_helper(machine_i2s_obj_t *self, size_t n_pos_args,
     // is SD valid?
     if (mp_obj_is_type(args[ARG_sd].u_obj, &pin_type)) {
         pin_af = pin_find_af(MP_OBJ_TO_PTR(args[ARG_sd].u_obj), AF_FN_I2S, self->i2s_id);
-        if (pin_af->type != AF_PIN_TYPE_I2S_SD) {
+        if (!(pin_af != NULL && (
+            pin_af->type == AF_PIN_TYPE_I2S_SD
+            #if defined(STM32H7)
+            || pin_af->type == AF_PIN_TYPE_I2S_SDO || pin_af->type == AF_PIN_TYPE_I2S_SDI
```

Comment:
Can SDO and SDI pins work in either direction?  if not, it would be good to validate the SDO and SDI versus RX vs TX mode.

Reply:
Yes, SDO and SDI can be swapped to work for either in or out.